src/Functions/FunctionsAES.h

```diff
@@ -409,12 +416,14 @@ template <typename Impl>
 class FunctionDecrypt : public IFunction
 {
 public:
+    explicit FunctionDecrypt(const ContextPtr & context) : use_null_when_decrypt_fail(AESHelpers::getParamsFromContext(context)) { }
     static constexpr OpenSSLDetails::CompatibilityMode compatibility_mode = Impl::compatibility_mode;
     static constexpr auto name = Impl::name;
-    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionDecrypt>(); }
+    static FunctionPtr create(ContextPtr context) { return std::make_shared<FunctionDecrypt>(context); }
```

Comment:
Is there a reason we can't just extract the setting right here without passing ContextPtr to the `FunctionDecrypt` constructor? 
```suggestion
    static FunctionPtr create(ContextPtr context)
    {
        return std::make_shared<FunctionDecrypt>(context->getSettingsRef().aes_decryption_use_null_when_fail);
    }
```

Reply:
It's my mistake. Before I used `context` in FunctionDecrypt but then remove it later.